### PR TITLE
gitfs: init at 0.2.5.

### PR DESCRIPTION
**Pull request by robbinch**
This adds gitfs, a FUSE filesystem that fully integrates with git.

### CODE REVIEW

File: pkgs/tools/filesystems/gitfs/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, python, buildPythonPackage, pythonPackages }:
+
+let inherit (stdenv.lib) getVersion versionAtLeast versionOlder; in
+
+assert versionAtLeast (getVersion pythonPackages.pygit2) "0.21.0";
```
Comment (lethalman) [inline pkgs/tools/filesystems/gitfs/default.nix]: Why this kind of versioning? If you need pygit2 between 0.21 and 0.22, then pass an appropriate pygit2 derivation.

**Comment by robbinch** [comment]
Ok, added derivations of specific versions of libgit2 and pygit2.

**Comment by lethalman** [comment]
Sorry what's the problem in using the already existing pygit2?

**Comment by robbinch** [comment]
There is no problem currently. But versions of gitfs work with specific versions of pygit2 (and libgit2). So if pygit2 is upgraded in future, this gitfs might not build. Hence, the constraints of versions.

**Comment by lethalman** [comment]
Then the required versions will be added in `python-packages.nix`, not in this package. Please remove this, in the future when it breaks, the old version will be added.

**Comment by robbinch** [comment]
Ok, have removed the constraints.

**Comment by lethalman** [comment]
Travis passed, thanks.

**Pull request merged**